scipy/optimize/_minimize.py

```diff
@@ -202,6 +204,17 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
     and either the Hessian or a function that computes the product of
     the Hessian with a given vector.
 
+    Method :ref:`trust-iterative <optimize.minimize-trustiterative>`
+    uses a iterative trust-region algorithm described in [13]_ for
+    unconstrained minimization.  This  algorithm requires
+    the gradient and Hessian; Recomended for medium size problems;
+    The advantage of this method is that it can provide a much more
+    accurate solution for the trust-region subproblem than, for instance,
+    the dogleg method.  For each iteration it solve the trust-region
+    subproblem iteratively requiring typically three Cholesky
+    factorizations (agaist the single Cholesky factorization needed by dogleg
+    method).
+
```

Comment:
The section should be expanded about the possibility to solve problems with indefinite Hessians. To be honest I don't have a good intuition about such problems, an example in the tutorial or small example in the docstring would be useful (and generally useful for sure). 

My understanding is that this property should help to go through regions with indefinite Hessian (with the help of trust-region bounds) and then reach a region where the Hessian is positive definite and a local minimum exists.